File: Java/Problem3.java
public class Problem3{
    private static final long NUM = 600851475143L;
    
     public static void main(String []args){
         
        //Initialize to true
        boolean[] isPrime = new boolean[(int)Math.floor(Math.sqrt(NUM)) + 1];
        for(int i = 2; i < isPrime.length; i++)
        	isPrime[i] = true;
        
        //Sieve algorithm, eliminating multiples
        for(int i = 2; i < isPrime.length; i++)
        	for(int j = i*2; j < isPrime.length; j+=i)
        		isPrime[j] = false;
        
		//Loop from the bottom down and find the largest prime multiple
        for(int i = isPrime.length - 1; i > 0; i--){
            if(isPrime[i] && NUM % i == 0){
                System.out.println(i);
                break;
            }
        }
	}
}

File: Python/Problem3.py
import math

bound = 600851475143
size = math.floor(math.sqrt(bound))
sieve = [True]*size

#Lets do sieve :)
for i in range(2, size - 1, 1):
	for j in range(i*2, size-1, i):
		sieve[j] = False

#Loop backwards and print largest prime that is a multiple of bound
for i in range(size - 1, 0, -1):
	if sieve[i] and bound % i == 0:
		print(i)
		break